shell/common/shell.cc

```diff
@@ -486,8 +500,17 @@ bool Shell::Setup(std::unique_ptr<PlatformView> platform_view,
   PersistentCache::GetCacheForProcess()->SetIsDumpingSkp(
       settings_.dump_skp_on_shader_compilation);
 
-  // Shell::Setup is running on the UI thread so we can do the following.
-  display_refresh_rate_ = weak_engine_->GetDisplayRefreshRate();
+  fml::AutoResetWaitableEvent ui_latch;
+
+  fml::TaskRunner::RunNowOrPostTask(
+      task_runners_.GetUITaskRunner(),
+      fml::MakeCopyable(
+          [engine = weak_engine_, &ui_latch,
+           &display_refresh_rate = this->display_refresh_rate_]() mutable {
+            display_refresh_rate = engine->GetDisplayRefreshRate();
+            ui_latch.Signal();
+          }));
+  ui_latch.Wait();
```

Comment:
@chinmaygarde thoughts on how to avoid the latching here?

Reply:
I don' think this that call should be on the engine to begin with. The vsync waiter has no thread affinity and there is no reason it has to accessed via the engine. I'd rather we used the `getUnsafe` call here with a TODO to correct vsync waiter ownership.